lib/ocpp/v201/charge_point.cpp

```diff
@@ -163,6 +163,23 @@ void ChargePoint::on_session_started(const int32_t evse_id, const int32_t connec
     this->evses.at(evse_id)->submit_event(connector_id, ConnectorEvent::PlugIn);
 }
 
+Get15118EVCertificateResponse
+ChargePoint::on_get_15118_ev_certificate_request(const Get15118EVCertificateRequest& request) {
+    EVLOG_debug << "Received Get15118EVCertificateRequest " << request;
+    auto future_res = this->send_async<Get15118EVCertificateRequest>(
+        ocpp::Call<Get15118EVCertificateRequest>(request, this->message_queue->createMessageId()));
+    const auto response_message = future_res.get();
+    EVLOG_debug << "Received Get15118EVCertificateResponse " << response_message.message;
+    if (response_message.messageType != MessageType::Get15118EVCertificateResponse) {
+        Get15118EVCertificateResponse response;
+        response.status = Iso15118EVCertificateStatusEnum::Failed;
+        return response;
+    }
```

Comment:
What could trigger this case? Regular errors, or a protocol violation by the CSMS?